internal/handler/msg_usersinfo.go

```diff
@@ -29,6 +38,240 @@ func (h *Handler) MsgUsersInfo(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
 		return nil, lazyerrors.Error(err)
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/1497
-	return nil, common.Unimplemented(document, document.Command())
+	dbName, err := common.GetRequiredParam[string](document, "$db")
+	if err != nil {
+		return nil, err
+	}
+
+	usersInfo, err := document.Get(document.Command())
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if usersInfo == nil {
+		msg := fmt.Sprintf("required parameter %q is missing", document.Command())
+		return nil, handlererrors.NewCommandErrorMsgWithArgument(handlererrors.ErrBadValue, msg, document.Command())
+	}
+
+	// TODO https://github.com/FerretDB/FerretDB/issues/3784
+	// TODO https://github.com/FerretDB/FerretDB/issues/3777
+	// TODO https://github.com/FerretDB/FerretDB/issues/3785
+	if err = common.UnimplementedNonDefault(document, "filter", func(v any) bool {
+		if v == nil || v == types.Null {
+			return true
+		}
+
+		cd, ok := v.(*types.Document)
+		return ok && cd.Len() == 0
+	}); err != nil {
+		return nil, err
+	}
+
+	common.Ignored(
+		document, h.L,
+		"showCredentials", "showCustomData", "showPrivileges",
+		"showAuthenticationRestrictions", "comment", "filter",
+	)
+
+	var (
+		users    []usersInfoPair
+		allDBs   bool // allDBs set to true means we want users from all databases
+		singleDB bool // singleDB set to true means we want users from a single database (when usersInfo: 1)
+	)
+
+	switch user := usersInfo.(type) {
+	case *types.Document:
+		if user.Has("forAllDBs") {
+			allDBs = true
+			break
+		}
+
+		var u usersInfoPair
+		if err = u.extract(user, dbName); err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		users = append(users, u)
+	case string:
+		var u usersInfoPair
+		if err = u.extract(user, dbName); err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		users = append(users, u)
+	case int, int32: // {usersInfo: 1 } // int for MongoDB and int32 for FerretDB
```

Comment:
It can't be `int`. You probably want `int64`.

Reply:
Oops. It's always int32. Removing `int`.